cassandra/src/main/java/com/yahoo/ycsb/db/CassandraCQLClient.java

```diff
@@ -195,12 +205,84 @@ public void init() throws DBException {
 
         session = cluster.connect(keyspace);
 
+        buildStatements();
+
       } catch (Exception e) {
         throw new DBException(e);
       }
     } // synchronized
   }
 
+  private void buildStatements() {
+    Properties p = getProperties();
+    int fieldCount = Integer.parseInt(
+            p.getProperty(CoreWorkload.FIELD_COUNT_PROPERTY, CoreWorkload.FIELD_COUNT_PROPERTY_DEFAULT));
+    String table = p.getProperty(CoreWorkload.TABLENAME_PROPERTY, CoreWorkload.TABLENAME_PROPERTY_DEFAULT);
+
+    fieldList = new ArrayList<String>(fieldCount);
+
+    // Insert and Update statement
+    final Insert is = QueryBuilder.insertInto(table);
+    is.value(YCSB_KEY, QueryBuilder.bindMarker());
+    fieldList.add(YCSB_KEY);
+
+    for (int i = 0; i < fieldCount; i++) {
+      // Preserve field iteration order for the HashMaps used in the DB methods
+      String field = FIELD_PREFIX + i;
+      fieldList.add(field);
+      is.value(field, QueryBuilder.bindMarker());
+    }
+    insertStatement = session.prepare(is);
+    insertStatement.setConsistencyLevel(writeConsistencyLevel);
+
+
+    // Delete statement
+    final Delete ds = QueryBuilder.delete().from(table);
+    deleteStatement = session.prepare(ds.where(QueryBuilder.eq(YCSB_KEY, QueryBuilder.bindMarker())));
+    deleteStatement.setConsistencyLevel(writeConsistencyLevel);
+
+    if (readallfields) {
+      // Select
+      String ss = QueryBuilder.select().all().from(table)
+              .where(QueryBuilder.eq(YCSB_KEY, QueryBuilder.bindMarker())).getQueryString();
+      selectStatement = session.prepare(ss);
+      selectStatement.setConsistencyLevel(readConsistencyLevel);
+
+      // Scan
+      String initialStmt = QueryBuilder.select().all().from(table).toString();
+      String scanStmt = getScanQueryString().replaceFirst("_", initialStmt.substring(0, initialStmt.length()-1));
+      scanStatement = session.prepare(scanStmt);
+      scanStatement.setConsistencyLevel(readConsistencyLevel);
+
+    } else {
+      selectStatements = new ConcurrentHashMap<String, PreparedStatement>(fieldCount);
```

Comment:
The `readallfields` boolean will build a `SELECT *` statement when true, otherwise the for-loop below would be used, and a `fieldCount == 1` would select just one field.  
Are you suggesting it is instead `if (readallfields || fieldCount == 1)`?